### ISSUE TITLE
Window menu for macOS user interface

### ISSUE THREAD

**Issue by rlaferla** [openscad/openscad#2674]
The macOS user interface desperately needs a Window menu to manage open windows.  It should contain "Minimize", "Zoom", "Bring All to Front" menu items.  If you open a .scad file that is already open, it needs to bring the already open file to the foreground and not open it a 2nd time.

![screen shot 2019-01-02 at 7 30 15 pm](https://user-images.githubusercontent.com/1135694/50619053-e501f380-0ec4-11e9-8026-9b34ac1c2b3a.png)

**Comment by Zenmaster43** [comment]
I concur with rlaferla on this. It's an incredible waste of time trying to locate the window I was just in before switching to another app and back, when I've got to multi-task on a major scale, with scores of windows open in each of many apps, spread over multiple spaces. I actually really like the way BBEdit does it, by using a single window for all documents and providing a document selection list in a bar on the left side of the window. This alleviates the need for a separate menu.

**Comment by Sod-Almighty** [comment]
I agree. On my system, I can minimise the active window by swiping down with four fingers on my Apple trackpad; or – and I believe this is a standard system hotkey – by pressing CMD+M. This works for 99% of applications. Guess which application it _doesn't_ work with?

The reason it doesn't work is almost certainly because there is no "Minimize" option in the Window menu. Is it really so much to ask for OpenSCAD to respect the fundamental user interface standards of the platform on which it is running?

**Comment by Sod-Almighty** [comment]
Why is this issue stale? Why is it stalled? It's not waiting on any information from the people who reported it. All it's waiting for is the developer to fix it.....after SEVEN BLOODY YEARS of being broken. Is it really too much to ask for the software to behave like any other MacOS application?

OpenSCAD is hard enough to use as it is, without putting extra usability hurdles between the software and the user.

**Comment by jordanbrown0** [comment]
Perhaps you misunderstand the relationship between the project and its developers.

Exactly zero people are paid to work on OpenSCAD.  There is no bug or enhancement request that is somebody's job to address.  Everything is done in somebody's spare time - and most or all of the active developers also have full-time paying jobs.

Anything that gets done, gets done because the developer personally wants it, or for fun, or out of pure altruism.  Apparently there is nobody to whom this is important enough to work on.

If this is so critical to you, please feel free to implement it and file a pull request.  Or pay somebody to do it.

But screaming at people who are giving you their work for free is not helpful.
